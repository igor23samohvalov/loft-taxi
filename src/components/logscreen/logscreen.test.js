import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import LogScreen from './logscreen.js';

describe('LogScreen', () => {
    it('renders correctly', () => {
        const { container } = render(<LogScreen initialPage='toAuthorization'/>)
        expect(container).toMatchSnapshot()
    });
    describe('when clicked on registration link', () => {
        it('switches to registration page', () => {
            const { getByText, getByTestId } = render(<LogScreen />)
            const regLink = getByText('Зарегистрируйтесь');
            fireEvent.click(regLink)
            expect(getByTestId('regTest')).toBeInTheDocument()
        })
    })
});