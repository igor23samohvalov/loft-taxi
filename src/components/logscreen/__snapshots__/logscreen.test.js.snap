// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogScreen renders correctly 1`] = `
<div>
  <div
    class="authorization-container"
    data-testid="authTest"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4"
    >
      Войти
    </h4>
    <form
      action=""
      data-id="toMap"
      href="#"
      id="authorization"
      required=""
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Новый пользователь? 
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          data-id="toRegistration"
          href="#registration"
        >
          Зарегистрируйтесь
        </a>
      </p>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          Имя пользователя *
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            placeholder="Имя пользователя"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          Пароль *
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            placeholder="Пароль"
            type="text"
            value=""
          />
        </div>
      </div>
    </form>
    <div
      class="button"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        form="authorization"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Войти
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
